### DISCUSSION TITLE
推荐 sublime text 作为编辑器，附上优缺点分析

### DISCUSSION THREAD

**Discussion by dreistein543** [nju-lug/NJUThesis#128] [Show and tell]
更换了sublime text 作为 LaTeX 编辑器，目前使用良好，与 v0.17.0 兼容良好

## 使用方法
- 安装 [sublime text](https://www.sublimetext.com/download) 并无限期试用（官方目前仍未强制收费）
- 安装 package control 并通过 `package control` 安装 `LaTeXTool` 、`LaTeX-cwl`、`AutoComplete` 插件
- 设定 latex 编译方式
- 直接套用 tex 文件写作
- 安装与设定方法可以参考 [论文环境搭建：MAC上的Sublime Text 3 + LaTex中文环境](https://zhuanlan.zhihu.com/p/61244778)
🌟特别是中文设定的 builder_setting 部分，可以完美解决毕业论文编译问题，实现 `ctrl/cmd + B`直接编译出PDF
>"program" : "xelatex",
>"command" : ["latexmk", "-cd", "-e", "$pdflatex = 'xelatex -interaction=nonstopmode -synctex=1 %S %O'", "-f", "-pdf"],

### 👉🏻M1 Mac 兼容性更好
TexStudio 虽然是全平台软件，但是一直没有出适配 M1 芯片 Mac 的官方版本（感兴趣同学可以自己查找[TexStudio官方wiki](https://github.com/texstudio-org/texstudio/wiki/Compiling)自行编译适配Apple Silicon芯片的版本，但是我尝试以后发现速度不如sublime）

### 👉🏻支持全平台正向/反向 `tex` 和 `pdf` 内容查找
在正反向查找这件事上，Windows 搭配 Smatra PDF 的体验是最好的，但是在非 windows 平台就比较麻烦，默认使用 skim（Mac平台）的 sublime LaTeXTool 能很好解决这一问题，在[LaTeXTool官方说明文档](https://latextools.readthedocs.io/en/latest/)直接查找 `inverse search` 可以看到详细说明。（似乎只要安装对应PDF阅读器也可以在其他编辑器中查找，不过不知道是否操作完全一致）

### 👉🏻能够实时预览数学公式
当然有许多同学是使用某些方法转化的 latex 语法，不过也有同学是纯手动书写（比如我）安装了 LaTeXTool 和 LaTeXXYZ 的sublime可以做到实时预览输入的公式，方便及时纠错（毕竟反向查找看到一堆代码也是头痛）

### 👉🏻参考文献引用更加清晰
之前在 #126 中我也描述了如何在 texstudio 中自动补全参考文献的citation key，但是考虑到这种自动补全只能让我们预览 citation key，并不能显示更多文献信息，特别是当出现 作者-年份 都相同的时候，citation key 的区别太小了，而在 sublime text 中搭配 latextool 引用文献可以清晰查看文献完整的 作者-标题-期刊-年份 等信息

### 👊🏻不能自动创建并跳转到不存在的文件
在 texstudio 中 如果输入 `\input{chapters/charpter1}` ，那么如果尚未创建 charpter1.tex 文件，可以自动创建并跳转，sublime则不行，或许可以通过某种插件实现？

### 👊🏻没有非常清晰的文件树
通过打开文件夹而非文件的方式却是可以引入文件树，但是主次关系不够明确，也不像texstudio可以通过内容来定义文件关系

### 👊🏻跨文件引用参考文献需要额外设置
⚠️ 如果是通过分章节tex文档的方式写论文，只在主文件（假设为 main.tex）中声明 bib 文档，那么需要在其他需要引用 参考文献的文档开头声明，来说明文档的主从关系（如果是习惯一开始就构建好所有文档的同学倒是不麻烦）
> % !TEX root = <main.tex的路径/main.tex>

**Comment by Yudezhi** [comment]
尚未在mac上测试，但是在Windows上，使用vscode可以直接进行预览数学公式，双向跳转，而查看引用文献更加方便由于没有这个需求未进行测试。

**Reply by dreistein543** [reply]
嗯嗯，vscode确实也是非常优秀的编辑器，各有所爱，只是我个人比较在意的是，vscode启动相对较慢，而且可选插件过多（当然这也是优势之一，只是我个人偏好更少的插件）。
